tests/test_postfix.py

```diff
@@ -0,0 +1,33 @@
+"""This module contains the tests for the postfix container, the image with postfix, sendmail and mailq pre-installed."""
+
+from bci_tester.data import POSTFIX_CONTAINERS
+
+CONTAINER_IMAGES = POSTFIX_CONTAINERS
+
+
+def test_postfix_status(auto_container):
+    # check if Postfix is running inside the container.
+    assert "localhost" in auto_container.connection.check_output(
+        "echo ${SERVER_HOSTNAME}"
+    )
+
+    assert (
+        "the Postfix mail system is running"
+        in auto_container.connection.check_output(
+            "postfix status > /tmp/status_output.txt 2>&1; cat /tmp/status_output.txt"
+        )
+    )
+
+
+def test_postfix_send_test_email(auto_container):
+    """send a test email using the Postfix container."""
+    sendmail_cmd = 'echo "Subject: Test Email\n\nThis is a test email body." | sendmail -v test@example.com'
+    auto_container.connection.check_output(sendmail_cmd)
+
+    # check the mail queue to ensure the test email was processed.
+    assert "Mail queue is empty" in auto_container.connection.check_output(
```

Comment:
how can that assert succeed given that there is no wait between sending the email and checking for the mail to be processed? isn't that inheritly racy?

and this works because test@example.com is locally delivered? or is it remotely delivered and these tests are spamming an external host?

Reply:
> and this works because [test@example.com](mailto:test@example.com) is locally delivered?

not working, it was failed delivery.